### PR TITLE
Extract reporters as separate artifacts

### PR DESCRIPTION
**Pull request by shlomisut**

### CODE REVIEW

File: metrics2-riemann-reporter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>com.yammer.metrics</groupId>
+            <artifactId>metrics-core</artifactId>
+            <version>2.1.2</version>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>riemann-java-client</artifactId>
+            <version>${project.version}</version>
```
Comment (shlomisut) [inline metrics2-riemann-reporter/pom.xml]: @royudassin does this do the trick ?

Comment (royudassin) [inline metrics2-riemann-reporter/pom.xml]: Yes. Great.

Best practice is to do the same for all external dependencies  - set their version inside <properties>

Comment (shlomisut) [inline metrics2-riemann-reporter/pom.xml]: @tarlog

File: metrics2-riemann-reporter/pom.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>metrics2-riemann-reporter</artifactId>
+    <parent>
+        <artifactId>riemann-java-client</artifactId>
+        <groupId>com.aphyr</groupId>
+        <version>0.3.1</version>
```
Comment (shlomisut) [inline metrics2-riemann-reporter/pom.xml]: @tarlog @royudassin should I change these too ?

Comment (tarlog) [inline metrics2-riemann-reporter/pom.xml]: You cannot. It's parent, you must specify it explicitly.

**Comment by shlomisut** [comment]
@tarlog Thanks for the help! If you can go over the final result It would be great

File: pom.xml [hunk 3]
```diff
@@ -1,180 +1,132 @@
...
+            </plugin>
 
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-all</artifactId>
-      <version>1.9.5</version>
-      <scope>test</scope>
-    </dependency>
-  </dependencies>
+            <plugin>
```
Comment (tarlog) [inline pom.xml]: Not sure when or why OSGi bundle is created, but from a glance, this should be inside riemann-java-client.

Comment (shlomisut) [inline pom.xml]: Well, I have no idea as to what OSGi is, but I'll check :)

File: pom.xml [hunk 4]
```diff
@@ -1,180 +1,132 @@
...
+                <configuration>
+                    <instructions>
+                        <Export-Package>
+                            com.aphyr.riemann;-noimport:=true,
+                            com.aphyr.riemann.client;-noimport:=true,
+                            com.codahale.metrics.riemann;-noimport:=true,
+                            com.yammer.metrics.reporting;-noimport:=true
+                        </Export-Package>
+                        <Import-Package>
+                            com.codahale.metrics;version="[3.0,4)";resolution:="optional",
```
Comment (shlomisut) [inline pom.xml]: @tarlog judging by the package names here - It actually seems to belong in the metrics modules - no ?

Comment (tarlog) [inline pom.xml]: Ah, well, it's kinda a mix now, after your refactoring.
This plugin creates an OSGi bundle.
So you declare which packages are exported and which are imported.
Export in OSGi means that other bundles (jars) can use only the packages that are declared as exported.
Imported is the opposite: which external packages are used by this bundle.
So you probably need to copy this plugin to each of the child modules and put the relevant packages..
The problem is that I'm not the OSGi expert and you need to run it in OSGi environment (e.g. Apache Felix) to make sure that your configurations are correct.

Comment (lielran) [inline pom.xml]: OSGi ?!!??!?! don't do that!

Comment (lielran) [inline pom.xml]: http://www.thoughtworks.com/radar/platforms/osgi 
HOLD

Comment (shlomisut) [inline pom.xml]: Not my project...

Comment (tarlog) [inline pom.xml]: Third party libraries usually provide OSGi support to be used in the OSGi containers (e.g. Eclipse, Websphere, Jboss, Glassfish use OSGi)
It doesn't mean that we need to use it here in Kenshoo, but it effects the pull requests to the third parties...

Comment (lielran) [inline pom.xml]: not true. 
Jboss ,Spring and other don't support OSGI no more. 
all using BOM for that ...

**Pull request merged**